### PR TITLE
Add `zip` stubs for 5 and more arguments.

### PR DESCRIPTION
**Pull request by sproshev**
Currently there are no stubs for zip with 5+ arguments.
So I added detailed stub for 5 arguments and very simple stub for 6+ arguments. I think it would be enough.

### CODE REVIEW

**Comment by sproshev** [comment]
Is it possible to fix `mypy` error? What is the better way to do it?

File: stdlib/2/__builtin__.pyi [hunk 1]
```diff
@@ -800,7 +801,15 @@ def zip(iter1: Iterable[_T1], iter2: Iterable[_T2],
...
-                                           _T3, _T4]]: ...  # TODO more than four iterables
+                                           _T3, _T4]]: ...
+@overload
+def zip(iter1: Iterable[_T1], iter2: Iterable[_T2], iter3: Iterable[_T3],
+        iter4: Iterable[_T4], iter5: Iterable[_T5]) -> List[Tuple[_T1, _T2,
+                                                                  _T3, _T4, _T5]]: ...
+@overload
+def zip(iter1: Iterable[Any], iter2: Iterable[Any], iter3: Iterable[Any],
+        iter4: Iterable[Any], iter5: Iterable[Any], iter6: Iterable[Any],
+        *iterables: Iterable[Any]) -> List[Tuple[Any, ...]]: ...
```
Comment (matthiaskramm) [inline stdlib/2/__builtin__.pyi]: Could we just do something like

```
@overload
def zip(*iter: Iterable) -> List[tuple]: ...
```

for the 5+ arg case?

Comment (sproshev) [inline stdlib/2/__builtin__.pyi]: I tried with

```
@overload
def zip(*iterables: Iterable[Any]) -> List[Tuple[Any, ...]]: ...
```

and it failed. See https://travis-ci.org/python/typeshed/jobs/212982069

**Comment by gvanrossum** [comment]
LGTM. Eventually this will be superseded by variadic type variables (https://github.com/python/typing/issues/193).

**Pull request merged**